### PR TITLE
Fix clickable tests (add interaction tests for focus and few behaviour tests)

### PR DESCRIPTION
**Pull request by hadasfa**

### CODE REVIEW

File: src/components/Clickable/Clickable.jsx [hunk 1]
```diff
@@ -41,11 +42,12 @@ const Clickable = forwardRef(
           disabled,
           [bemHelper({ state: "disable-text-selection" })]: !enableTextSelection
         })}
+        data-testid={ELEMENT_TYPES.CLICKABLE}
         role={role}
         onClick={disabled ? undefined : onClick}
         id={id}
         onKeyDown={disabled ? undefined : onKeyDown}
-        tabIndex={tabIndex}
+        tabIndex={disabled ? -1 : tabIndex}
```
Comment (hadasfa) [inline src/components/Clickable/Clickable.jsx:50]: fix bug - we allowed focus on disabled clickable element

**Review by laviomri — COMMENTED** [review]
package-lock shouldn't have any changes - something's probably wrong @hadasfa

File: src/components/Clickable/__stories__/Clickable.stories.mdx [hunk 2]
```diff
@@ -1,12 +1,16 @@
 import Clickable from "../Clickable";
 import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
-import { createComponentTemplate } from "../../../storybook/functions/create-component-story";
-import "./clickable.stories.scss";
 import {
   BUTTON,
   HIDDEN_TEXT,
   ICON_BUTTON
-} from "../../../storybook/components/related-components/component-description-map";
+} from "../../../storybook/components/related-components/component-description-map"; import {
```
Comment (laviomri) [inline src/components/Clickable/__stories__/Clickable.stories.mdx:7]:
```suggestion
} from "../../../storybook/components/related-components/component-description-map"; 
import {
```

File: src/components/Clickable/__stories__/Clickable.stories.mdx [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
+    StoryDescription
+} from "../../../storybook/components/story-description/story-description";
+import {Flex} from "components";
+import "./clickable.stories.scss"; import {
```
Comment (laviomri) [inline src/components/Clickable/__stories__/Clickable.stories.mdx:11]:
```suggestion
import "./clickable.stories.scss"; 
import {
```

File: src/components/Clickable/__stories__/Clickable.stories.mdx [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+    statesPlaySuite
+} from "components/Clickable/__tests__/clickable.interactions";
```
Comment (laviomri) [inline src/components/Clickable/__stories__/Clickable.stories.mdx:13]: WDYT about a relative import instead? this is how we do it elsewhere
```suggestion
} from "../__tests__/clickable.interactions";
```

File: src/components/Clickable/__tests__/clickable.interactions.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+async function states_onClickTabFocusElementTest(canvas) {
+  const CLICKABLE_LABEL = "clickable button";
+  const CLICKABLE_DISABLED_LABEL = "disabled clickable button";
+  const clickableElement = await getByLabelText(canvas, CLICKABLE_LABEL);
+  const disabledClickableElement = await getByLabelText(canvas, CLICKABLE_DISABLED_LABEL);
+  await pressNavigationKey(NAVIGATIONS_COMMANDS.TAB);
+  expect(document.activeElement).toEqual(clickableElement);
+  await pressNavigationKey(NAVIGATIONS_COMMANDS.TAB);
+  expect(document.activeElement).not.toEqual(disabledClickableElement);
```
Comment (laviomri) [inline src/components/Clickable/__tests__/clickable.interactions.js:18]: what would you expect `document.activeElement` to be here? undefined?

File: src/__tests__/interactions-helper.js [hunk 6]
```diff
@@ -84,6 +84,18 @@ export const typeText = async (element, text, waitForDebounceMs = 250) => {
   return result;
 };
 
+export const pressNavigationKey = async (command = NAVIGATIONS_COMMANDS.TAB, waitForDebounceMs = 250) => {
```
Comment (laviomri) [inline src/__tests__/interactions-helper.js:87]: personally I think that `waitForDebounceMs` should be 0 by default. I think most cases where we navigate we don't have a debounce. Usually when we navigate it's not like we trigger an input, which might require a debounce... but I might be wrong, you're more familiar than me with the components :)

File: src/components/Clickable/__tests__/clickable-interactions-tests.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  await pressNavigationKey(NAVIGATIONS_COMMANDS.TAB);
+  console.log(document.activeElement);
+  expect(document.activeElement).not.toEqual(disabledClickableElement);
+  expect(canvas).not.toContain(document.activeElement);
+}
+
+export const statesPlaySuite = interactionSuite({
+  tests: [states_onClickTabFocusElementTest],
+  afterEach: async () => {
+    debugger;
```
Comment (laviomri) [inline src/components/Clickable/__tests__/clickable-interactions-tests.js:26]: 😱

**Review by laviomri — APPROVED** [review]
don't forget the package-lock :D

**Pull request merged**